### PR TITLE
contrib: snapshot to checkpoint converter

### PR DESCRIPTION
**Pull request by ibhatt-jumptrading**

### CODE REVIEW

File: src/flamenco/runtime/program/fd_bpf_program_util.c [hunk 1]
```diff
@@ -204,12 +204,15 @@ fd_bpf_scan_and_create_bpf_program_cache_entry( fd_exec_slot_ctx_t * slot_ctx,
   fd_funk_txn_t * parent_txn = slot_ctx->funk_txn;
   slot_ctx->funk_txn = cache_txn;
 
+  FD_LOG_NOTICE(("VALUE"));
+  ulong value = 0;
```
Comment (lheeger-jump) [inline src/flamenco/runtime/program/fd_bpf_program_util.c:208]: can you remove now?

File: src/flamenco/runtime/program/fd_bpf_program_util.c [hunk 2]
```diff
@@ -204,12 +204,15 @@ fd_bpf_scan_and_create_bpf_program_cache_entry( fd_exec_slot_ctx_t * slot_ctx,
...
   for (fd_funk_rec_t const *rec = fd_funk_txn_first_rec( funk, funk_txn );
        NULL != rec;
        rec = fd_funk_txn_next_rec( funk, rec )) {
     if( !fd_funk_key_is_acc( rec->pair.key ) ) {
       continue;
     }
+    value += fd_funk_val_sz( rec );
```
Comment (lheeger-jump) [inline src/flamenco/runtime/program/fd_bpf_program_util.c:215]: ditto

File: src/flamenco/runtime/program/fd_bpf_program_util.c [hunk 3]
```diff
@@ -233,6 +236,7 @@ fd_bpf_scan_and_create_bpf_program_cache_entry( fd_exec_slot_ctx_t * slot_ctx,
 
     cnt++;
   }
+  FD_LOG_NOTICE(("VALUE: %lu", value));
```
Comment (lheeger-jump) [inline src/flamenco/runtime/program/fd_bpf_program_util.c:239]: ditto

**Review by lheeger-jump — CHANGES_REQUESTED** [review]

File: src/flamenco/runtime/program/fd_bpf_program_util.c (hunk 3 above)
Comment (ibhatt-jumptrading) [reply src/flamenco/runtime/program/fd_bpf_program_util.c:239]: These shouldn't have gone in... my mistake

File: contrib/checkptify/checkptify.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        checkpt_files.append( file )
+
+# Remove all but the newest N checkpoints
+checkpt_files.sort( key=os.path.getmtime, reverse=True)
+[ os.remove( output_dir + file ) for file in checkpt_files[2:] ]
+
+# Download an incremental snapshot
+full_command = "wget --trust-server-names " + solana_endpoint + INCREMENTAL_ENDPOINT
+print( full_command )
+subprocess.run( full_command, shell = True )
```
Comment (lidatong) [inline contrib/checkptify/checkptify.py:63]: you can use a list instead of `+` for readability

File: contrib/checkptify/checkptify.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+os.chdir( firedancer_dir )
+full_snapshot_path        = output_dir + full_snapshot_file
+incremental_snapshot_path = output_dir + incremental_snapshot_file
+checkpt_path              = output_dir + str(incremental_end_slot) + "-checkpt"
+obj_dir = os.environ["OBJDIR"]
+ingest_command = "./" + obj_dir + "/bin/fd_ledger --cmd ingest --snapshot " + full_snapshot_path + \
+                 " --incremental " + incremental_snapshot_path + " --funk-only 1 " + \
+                 " --checkpt-funk " + checkpt_path + " --funk-page-cnt " + str(num_pages) + \
+                 " --index-max " + str(index_max)
```
Comment (lidatong) [inline contrib/checkptify/checkptify.py:98]: use a list

**Review by lidatong — APPROVED** [review]

**Review by lheeger-jump — APPROVED** [review]

**Pull request merged**